### PR TITLE
Feature/admin reorder for logged in customer

### PR DESCRIPTION
**Pull request by johnmccombs1**
# Re-order feature for logged-in customer in Admin panel on Magento 2

**Here is the list PRs of child tasks that were approved**
1. Create schemas for generating a new table for saving the customer credit card information
https://github.com/BoltApp/bolt-magento2/pull/436
2. Create model, resource model and resource model collection for the credit card entity 
https://github.com/BoltApp/bolt-magento2/pull/437
3. Create a new credit card for the customer when the pending hook is sent to  Magento
https://github.com/BoltApp/bolt-magento2/pull/440
4. Ignore the credit card creation logic if the cart exists
https://github.com/BoltApp/bolt-magento2/pull/441
https://github.com/BoltApp/bolt-magento2/pull/480
5. Render the credit card list in a selection box when the administrator chooses Bolt as the payment 
https://github.com/BoltApp/bolt-magento2/pull/443
https://github.com/BoltApp/bolt-magento2/pull/478
6. Toggle Bolt checkout button if the administrator chooses to use a new  or an existing credit card
https://github.com/BoltApp/bolt-magento2/pull/446
7. Re-charge the customer (send a re-charge API call to Bolt) if the administrator uses an existing credit card then submits the order)
https://github.com/BoltApp/bolt-magento2/pull/451
8. List all credit cards in the admin customer detail page
https://github.com/BoltApp/bolt-magento2/pull/459
9. Remove credit card  in the admin customer detail page
https://github.com/BoltApp/bolt-magento2/pull/479
10. List all credit cards  in the admin customer detail page 
https://github.com/BoltApp/bolt-magento2/pull/488
11. Remove credit card  in the customer detail page
https://github.com/BoltApp/bolt-magento2/pull/489

Ansa task link: https://app.asana.com/0/564264490825835/1150842025023777

# Type of change

- [ ] Bug fix (change which fixes an issue)
- [x] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please validate that you have tested your change in at least one of the following areas:

- [x] Successfully tested locally (or docker image)
- [x] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] New and existing unit tests pass locally with my changes.
- [x] I have created or modified unit tests to sufficiently cover my changes.
- [x] I have added my Asana task link and provided a changelog message if applicable.

### CODE REVIEW

**Comment by matt-thomason** [comment]
Is there a diff between "PRs that were reviewed as part of merging into the feature branch" and "code in this branch that haven't been reviewed"?

File: Model/Api/OrderManagement.php [hunk 1]
```diff
@@ -77,11 +77,6 @@ class OrderManagement implements OrderManagementInterface
      */
     private $configHelper;
 
-    /**
-     * @var CartHelper
-     */
-    private $cartHelper;
```
Comment (danac-gs) [inline Model/Api/OrderManagement.php]: Deleting this seems to be a mistake

File: Model/CustomerCreditCard.php [hunk 2]
```diff
@@ -45,7 +45,7 @@ class CustomerCreditCard extends AbstractModel implements \Magento\Framework\Dat
     private $dataObjectFactory;
 
     /**
-     * @var ApiHelper
+     * @var ApiHelperd
```
Comment (danac-gs) [inline Model/CustomerCreditCard.php]: typo

**Comment by coredevelop** [comment]
> LGTM, with one note. There are changes marked here that are not related to this task, from another branch I guess. It might be an error on github end. However, did you merge the latest master into your branch?

@danac-gs Yes, I did.

**Comment by johnmccombs1** [comment]
> LGTM, with one note. There are changes marked here that are not related to this task, from another branch I guess. It might be an error on github end. However, did you merge the latest master into your branch?

@danac-gs Yes, I did.

**Comment by johnmccombs1** [comment]
> Is there a diff between "PRs that were reviewed as part of merging into the feature branch" and "code in this branch that haven't been reviewed"?

@matt-thomason I merged the master branch into my branch, fixed the conflicts and reviewed them already.

**Comment by matt-thomason** [comment]
@johnmccombs1 Sorry this has been hanging out. Can you rebase this today, and we can get this merged? Ping me when it's done.

**Comment by johnmccombs1** [comment]
@matt-thomason  I've re-based the branch. Can you re-check?

**Pull request merged**